streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -204,16 +206,17 @@ public void register(final StateStore store,
                                                              final List<ConsumerRecord<byte[], byte[]>> records) {
         final long limit = offsetLimit(storePartition);
         List<ConsumerRecord<byte[], byte[]>> remainingRecords = null;
+        final List<KeyValue<byte[], byte[]>> restoreRecords = new ArrayList<>();
 
         // restore states from changelog records
-        final StateRestoreCallback restoreCallback = restoreCallbacks.get(storePartition.topic());
+        final BatchingStateRestoreCallback restoreCallback = getBatchingRestoreCallback(restoreCallbacks.get(storePartition.topic()));
 
         long lastOffset = -1L;
         int count = 0;
         for (final ConsumerRecord<byte[], byte[]> record : records) {
             if (record.offset() < limit) {
                 try {
-                    restoreCallback.restore(record.key(), record.value());
+                    restoreRecords.add(KeyValue.pair(record.key(), record.value()));
```

Comment:
Is the `try-catch` still needed here? And is it possible that some `Exception` be thrown from `restoreCallback.restoreAll` below? If yes we need to think if we want to handle it here, or in higher-levels of the call trace.